### ISSUE TITLE
Sending Array to Zabbix API 

### ISSUE THREAD

**Issue by qk4l** [lukecyca/pyzabbix#70]
Hi,

I'm playing around [trigger.deletedependencies](https://www.zabbix.com/documentation/2.4/manual/api/reference/trigger/deletedependencies) call and try to send array of triggerid but seems pyzabbix does not allow to do it in properly way.

**Does not work**
```
     hosts_tr = zapi.trigger.get(group=hostgroup, filter={'description': trigger_desc})
     # Short version of hosts_tr:  [{u'triggerid': u'24344'}, {u'triggerid': u'24352'}]
     zapi.trigger.deletedependencies(hosts_tr)
```
**Output**
```
Sending: {
    "params": [
        [
            {
                "triggerid": "24344"
            },
            {
                "triggerid": "24352"
            }
        ]
    ],
    "jsonrpc": "2.0",
    "method": "trigger.deletedependencies",
    "auth": "My auth here",
    "id": 71
}
Response Code: 200
Response Body: {
    "jsonrpc": "2.0",
    "result": {
        "triggerids": []
    },
    "id": 71
}
```

request_json was wrapped array in additional list so Zabbix did not give my ids :(

**Works**
```
     zapi.trigger.deletedependencies({u'triggerid': u'24344'}, {u'triggerid': u'24352'})
```
I guest it may affect any methods that waiting array of objects.

**Comment by neothematrix** [comment]
hi, you raised a very interesting question here.
It turns out that there are some kind of API requests that behave differently than the others, like in your example, the "triggers.deletedependencies" (but the same goes for triggers.delete and a few others), if you look at the doc, this method wants a list/array of arguments without any principal element, in the form of:
```
"value1", "value2", "value3"
```
while all the other methods expect always a principal element, in the form of:
```
parameter1: {"value1", "value2", "value3"}, parameter2: {"value1","value2","value3"}
```
to make a long story short, the dynamic creation of the request implemented on pyzabbix can't handle this correctly and I don't think there's any way to deal with it in a nice way (please correct me if I'm wrong). This is rather something to ask Zabbix to fix, as these api methods are behaving differently than all the others.
Anyway, what you would need to workaround this, is basically to transform your parameter into an "arg" by appending the "*" to your passed parameter:

```
hosts_tr = zapi.trigger.get(group=hostgroup, filter={'description': trigger_desc})
# Short version of hosts_tr:  [{u'triggerid': u'24344'}, {u'triggerid': u'24352'}]
zapi.trigger.deletedependencies(*hosts_tr)
```

please try and let me know if it works, and please, if anybody has any suggestion on how to deal with this, let me know that too!

**Comment by qk4l** [comment]
> Anyway, what you would need to workaround this, is basically to transform your parameter into an "arg" by appending the "*" to your passed parameter:

Thanks for workaround. It works.

I think it's a good idea to create issue in Zabbix tracker but I'm not sure that I can describe issue in detail from API side.

**Comment by ironbishop** [comment]
Please document this workaround in the README... it affects the dashboard.delete method too.
